openedx/stanford/cms/djangoapps/contentstore/views/utilities/tasks.py

```diff
@@ -0,0 +1,98 @@
+"""
+Perform bulk operations on course problems via asynchronous tasks
+"""
+
+import logging
+
+from celery import task
+from django.conf import settings
+from django.core.mail import send_mail
+from smtplib import SMTPException
+from django.core.urlresolvers import reverse
+
+from edxmako.shortcuts import render_to_string
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+
+from cms.djangoapps.cms_user_tasks.tasks import send_task_complete_email
+
+log = logging.getLogger('edx.celery.task')
+
+
+def _send_email_on_completion(course, user_fields, modified_settings, success):
+    """
+    Send email to user on completion of task, either success or failure
+    """
+    task_name = 'bulk settings update'
+    task_state_text = 'success'
+    dest_addr = user_fields['email']
+    try:
+        detail_url = reverse('utility_bulksettings_handler',kwargs={'course_key_string': course.id,})
```

Comment:
If we're going to use the `reverse_course_url` (or whatever) helper, we should use it everywhere.